### PR TITLE
[feature-1091]: Update Authorization Proxy Server Ingress and Use Local Storage for Redis by default

### PR DESCRIPTION
**Pull request by atye**
# Description

This updates the Authorization proxy-server ingress to enable connecting by a cluster node IP address instead of requiring one of the hostnames in the ingress, such as csm-authorization.com, because there is extra work to make a hostname like that resolve correctly. It also provisions a local volume for Redis by default if no storage class for Redis is specified.

Changes in some more detail:

Adds v1.10.0 folder for Authorization.

`operatorconfig/moduleconfig/authorization/v1.10.0/ingress.yaml` contains the following lines to allow connecting by IP address (no `host` specified).
```
 - http:
      paths:
      - backend:
          service:
            name: proxy-server
            port:
              number: 8080
        path: /
        pathType: Prefix
```

Adds `operatorconfig/moduleconfig/authorization/v1.10.0/local-provisioner.yaml` which contains manifests for a local provisioning storage class and a persistent volume with said storage class. If the `REDIS_STORAGE_CLASS` env variable is not provided, these resources will be created and used for Redis.

Adds sample file for Authorization v1.10.0 where `REDIS_STORAGE_CLASS` is not specified by default.

Adds unit test for deploying Authorization with default local storage for Redis.

Adds PROXY_HOST env variable to `tests/e2e/env-e2e-test.sh` to specify the IP or hostname of the proxy-server for karavictl instead of using hardcoded address that only resolves in a k8s cluster.

Disabled cert-manager dependency in `tests/e2e/testfiles/authorization-templates/csm_authorization_proxy_server.yaml` because a prerequisite step of Install Authorization Proxy Server is creating a cert which will fail unless cert-manager is installed. So, cert-manager has to be installed before e2e and shouldn't be installed again with Authorization.

Adds e2e scenario for installing Authorization using default, local storage.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1091 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

New unit test and manual e2e test.

### CODE REVIEW

**Review by xuluna — DISMISSED** [review]

File: tests/e2e/testfiles/authorization-templates/csm_authorization_proxy_server_default_redis.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+          # Default value: nginx
+          - name: "PROXY_INGRESS_CLASSNAME"
+            value: "nginx"
+          # An additional host rule for the proxy-server ingress
+          # Default value: authorization-ingress-nginx-controller.namespace.svc.cluster.local
+          - name: "PROXY_INGRESS_HOST"
+            value: "authorization-ingress-nginx-controller.authorization.svc.cluster.local"
+
+          # Specify storage class for redis. Otherwise, default storage class is used.
+          # Default value: None
```
[the remark this answers is not in the document]
Comment (atye) [reply tests/e2e/testfiles/authorization-templates/csm_authorization_proxy_server_default_redis.yaml:47]: Updated.

File: pkg/modules/testdata/cr_auth_proxy_no_redis.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          # Default value: nginx
+          - name: "PROXY_INGRESS_CLASSNAME"
+            value: "nginx"
+          # An additional host rule for the proxy-server ingress
+          # Default value: authorization-ingress-nginx-controller.namespace.svc.cluster.local
+          - name: "PROXY_INGRESS_HOST"
+            value: "authorization-ingress-nginx-controller.authorization.svc.cluster.local"
+
+          # Specify storage class for redis. Otherwise, default storage class is used.
+          # Default value: None
```
[the remark this answers is not in the document]
Comment (atye) [reply pkg/modules/testdata/cr_auth_proxy_no_redis.yaml:46]: Updated.

File: tests/e2e/testfiles/values.yaml [hunk 3]
```diff
@@ -13,6 +13,20 @@
     - "Configure authorization-proxy-server for [powerflex]"
     - "Delete custom resource [1]"
 
+- scenario: "Install Authorization Proxy Server"
```
[the remark this answers is not in the document]
Comment (atye) [reply tests/e2e/testfiles/values.yaml]: Updated.

File: samples/authorization/csm_authorization_proxy_server_v1100.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        storageService: dellemc/csm-authorization-storage:v1.9.0
+        redis: redis:6.0.8-alpine
+        commander: rediscommander/redis-commander:latest
+        opa: openpolicyagent/opa
+        opaKubeMgmt: openpolicyagent/kube-mgmt:0.11
+        envs:
+          # base hostname for the ingress rules that expose the services
+          # the proxy-server ingress will use this hostname
+          # Allowed values: string
+          # Default value: csm-authorization.com
```
[the remark this answers is not in the document]
Comment (atye) [reply samples/authorization/csm_authorization_proxy_server_v1100.yaml]: Updated.

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by PeresKereotubo — APPROVED** [review]

**Review by EvgenyUglov — APPROVED** [review]

**Pull request merged**